enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,501 @@
+- Functionality within the CCM to create and manage the lifecycle of AWS Security Groups for NLBs, including creating ingress and egress rules based on the service specification. This would likely involve using the AWS SDK for Go to interact with the EC2 API for creating and managing security groups.
+
+#### Cluster Ingress Operator (CIO)
+
+- FeatureGate TechPreview (TODO describe the name?)
+- Introduce a new boolean field `managedSecurityGroup` in the `NetworkLoadBalancer` provider parameters within the IngressController API (`operator.openshift.io/v1`).
+
+```go
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer. For Example: Setting AWS EIPs https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+type AWSNetworkLoadBalancerParameters struct {
+	// managedSecurityGroup specifies whether the service load balancer should create
+	// and manage security groups for the Network Load Balancer.
+	//
+	// +optional
+	// +openshift:enable:FeatureGate=IngressNLBSecurityGroup
+	ManagedSecurityGroup bool `json:"managedSecurityGroup,omitempty"`
+}
+```
+
+- The CIO will set the `service.beta.kubernetes.io/aws-load-balancer-managed-security-group: "true"` annotation on the default router Service if the `managedSecurityGroup` field in the IngressController spec is set to true.
+
+#### Installer
+
+- Introduce a new boolean field `managedSecurityGroup` under `platform.aws.ingressController` in the `install-config.yaml`. This field will only be considered when `platform.aws.lbType` is `NLB`.
+
+```go
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+	// IngressController is an optional extra configuration for the Ingress Controllers.
+	// +optional
+	IngressController *IngressController `json:"ingressController,omitempty"`
+}
+
+// IngressController specifies the additional ingress controller configuration.
+type IngressController struct {
+	// SecurityGroupEnabled is an optional field to enable security groups
+	// when using load balancer type Network Load Balancer (NLB).
+	// When this field is enabled with LBType NLB, all ingresscontrollers (including the
+	// default ingresscontroller) will be created using security group in the NLBs
+	// by default.
+	//
+	// If this field is not set explicitly, it defaults to no security groups in the NLB.
+	// This default is subject to change over time.
+	//
+	// +optional
+	SecurityGroupEnabled bool `json:"securityGroupEnabled,omitempty"`
+}
+```
+
+- TBD: deprecate the field `platform.aws.lbType` in favor of `platform.aws.ingressController.loadBalancerType`
+
+#### ROSA Classic
+
+- TBD: API changes in Hive to read and process the new install-config option.
```

Comment:
Hive itself _should not_ need any changes, as we consume the install-config.yaml as a black box. I think Clusters Service is the component that would deal with this.

That said, it is customary for us to create a QE-only HIVE card to be executed once the upstream work is done, just to make sure we didn't miss anything.